### PR TITLE
RHEL 8 support

### PR DESCRIPTION
**Pull request by kbgl**
For usage please look into README.rhel.md. I'm still extending README.rhel.md, we can later join the content into README.md. For a long term we can drop pelion-build-all.sh and docker-run.sh and use build-all.sh and docker-run-env.sh instead.

Currently supported: 
- RHEL 8 build for amd64 and arm64 (requires subscription)
- centos 8 build for amd64 and arm64 (packages build on Centos can be installed on RHEL)
- docker-run-env.sh supports environment for amd64 and arm64
- scripts can reuse container to speed up build (very userful for arm64 on amd64 build)
- images are created automatically when needed

General idea was to separate environment setup from build scripts, and reuse docker function to build single package, all package or run shell for selected environment.

According to RHEL policy all services are disabled after installation so required services has to be enabled. Dependent services are started automatically, eg. enabling devicedb will start edge-core, wait-for-pelion-identity, maestro. 

@kunal11191 please run tests, at the moment I run simple tests, but Kubelet is not well tested - service starts and log output looks similar to Debian's log.

@ryannowarm blocking wait-for-pelion-identity.service functionality is implemented in d893b52, 0ba03a5 and 63c9296. In short: wait-for-pelion-identity.service type is 'oneshot', and the script called there does not exit until we generate identity - which blocks startup of other pelion services. Seems to boot up faster than 'restart' approach.

It is not yet ready for merge still needs:
- [x] cleanup - remove testfiles etc
- [x] metapackages - I'm still investigating this; it looks like this concept is different in RHEL
- [x] extend README.rhel.md

Environment for Ubuntu Focal was also created, I will update it and push after closing this PR. Similarly we can port all debian-like distros to new scripts

Please let me know if you have any problems with the scripts.

### CODE REVIEW

**Comment by kbgl** [comment]
@kunal11191 there are build failures shown in *checks* section above, are those related to RHEL?

**Review by kunal11191 — COMMENTED** [review]
> @kunal11191 there are build failures shown in _checks_ section below, are those related to RHEL?

@kbgl right now PR checks are not working on RHEL but on older builds bionic, focal, buster (arm64, armhf, amd64).
Above mentioned builds on dev are failing with below given error message

`Generating source package of 'maestro'
INFO: Source preparation!
Cloning into '/pelion-build/build/downloads/maestro/maestro'...
Fetching origin
Note: checking out '20caa5d032424a11146b7923eaaed74e80de96da'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 20caa5d Add aarch64
INFO: Generating Debian source packages!
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: using patch list from debian/patches/series
dpkg-source: info: applying greaselib-autoreconf.diff
dpkg-source: info: applying gperftools-enable-unwind.diff
dpkg-source: info: applying eventfd.diff
dpkg-source: error: LC_ALL=C patch -t -F 0 -N -p1 -u -V never -E -b -B .pc/eventfd.diff/ --reject-file=- < /pelion-build/build/tmp-build/maestro/maestro-0.0.1/debian/patches/eventfd.diff subprocess returned exit status 1
patching file processes/wakeupfd.go
Reversed (or previously applied) patch detected!  Skipping patch.
1 out of 1 hunk ignored
dpkg-source: info: the patch has fuzz which is not allowed, or is malformed
dpkg-source: info: if patch 'eventfd.diff' is correctly applied by quilt, use 'quilt refresh' to update it
script returned exit code 2`

File: README.rhel.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+SELinux is not currently supported: `pelion-relay-term`  service will be killed when SELinux is enabled.
+SELinux has to be disabled, set to permissive or `node` binary should be excluded. More details about SELinux and its configuration here: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/using_selinux/index
+
+## Target package installation
+1. Before installing packages make sure that you have subscription enabled and EPEL, CodeReady and Docker repositories are enabled (see [RHEL repositories](#rhel-Repositories) and [Docker](#docker)).
+2. Copy content of `build/deploy/rpm/<distro>/<arch>` and `build/deploy/rpm/<distro>/noarch/` to target system (where `<arch>` is `amd64` or `arm64` and `<distro>` is `rhel8` or `centos8`).
+
+3. To install use `yum` command, for example if all packages are in current directory run:
+```bash
+sudo yum install *.rpm
```
[the remark this answers is not in the document]
Comment (kbgl) [reply README.rhel.md:130]: yes, we can use `-y`. Now i'm working on unification of README.md and README.rhel.md, this file will be removed in next PR.

**Comment by kbgl** [comment]
> Cloning into '/pelion-build/build/downloads/maestro/maestro'...
> Fetching origin
> Note: checking out '20caa5d032424a11146b7923eaaed74e80de96da'.

Looks like problem in CI. The change with hash `20caa5d032424a11146b7923eaaed74e80de96da` was merged to `dev` branch 
 but not here yet, it should not be visible in log here.

**Comment by kunal11191** [comment]
> Looks like problem in CI. The change with hash `20caa5d032424a11146b7923eaaed74e80de96da` was merged to `dev` branch
> but not here yet, it should not be visible in log here.

since redhat-83-support is supposed to be merged to dev it will take latest version of dev add your code to it and then trigger builds and tests.
So that you know in advance will your change break the dev or master

**Comment by kbgl** [comment]
> > Looks like problem in CI. The change with hash `20caa5d032424a11146b7923eaaed74e80de96da` was merged to `dev` branch
> > but not here yet, it should not be visible in log here.
> 
> since redhat-83-support is supposed to be merged to dev it will take latest version of dev add your code to it and then trigger builds and tests.
> So that you know in advance will your change break the dev or master

Thanks for the explanation.

**Comment by costanic** [comment]
[Here is a PR to fix dev branch.](https://github.com/PelionIoT/distro-pelion-edge/pull/123)

**Review by ryannowarm — APPROVED** [review]

**Comment by ryannowarm** [comment]
@kbgl , we need CI to pass before merging this.  I'm trying to retrigger the CI without much luck.  I've opened a bug for the SQA team on this: https://go.arm.com/X5mPLLW

FYI: @mohitsingalarm and @kunal11191

**Review by kunal11191 — APPROVED** [review]
Approving this am able to build required files for both rhel and centos on arm64 and amd64; also able to install and run smoke suite

**Pull request merged**